e2e-tests/test/hardhat-apis.test.ts

```diff
@@ -64,6 +64,24 @@ describe("hardhat_mine", function () {
   });
 });
 
+describe("hardhat_setNextBlockBaseFeePerGas", function () {
+  it("should allow setting the next block's base fee per gas", async function () {
+    // Arrange
+    const price = "0x12d687";
+
+    // Act
+    const previous = await provider.send("eth_getBlockByNumber", ["latest", false]);
+    await provider.send("hardhat_setNextBlockBaseFeePerGas", [price]);
+    await provider.send("hardhat_mine");
+    const latest = await provider.send("eth_getBlockByNumber", ["latest", false]);
+    await provider.send("hardhat_setNextBlockBaseFeePerGas", [previous.baseFeePerGas]);
```

Comment:
why do we do this call again?

Reply:
To leave the test environment as we found it - letting a test change a fee might affect other tests (like the `hardhat_impersonateAccount` case).